src/realm_dart.hpp

```diff
@@ -16,15 +16,24 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 
-#ifndef REALM_DART_SYNC_CLIENT_CONFIG_H
-#define REALM_DART_SYNC_CLIENT_CONFIG_H
+#pragma once
 
-#include "realm.h"
+#include "realm_dart.h"
+#include <realm/object-store/c_api/types.hpp>
 
-RLM_API void realm_dart_sync_client_config_set_log_callback(realm_sync_client_config_t* config, 
-                                                            realm_log_func_t callback, 
-                                                            void* userdata,
-                                                            realm_free_userdata_func_t userdata_free,
-                                                            realm_scheduler_t* scheduler) RLM_API_NOEXCEPT;
+struct realm_dart_userdata_async {
+public:
+    realm_dart_userdata_async(Dart_Handle handle, void* callback, realm_scheduler_t* scheduler)
+    : handle(Dart_NewPersistentHandle_DL(handle))
+    , dart_callback(callback)
+    , scheduler(*scheduler)
+    { }
 
-#endif // REALM_DART_SYNC_CLIENT_CONFIG_H
+    ~realm_dart_userdata_async() {
+        Dart_DeletePersistentHandle_DL(handle);
+    }
+
+    Dart_PersistentHandle handle;
```

Comment:
So we always take persistent handle. We will need to revisit how how to handle users who forget to cancel their stream subscriptions, but I think we can handle that more elegantly on the dart side with the `WeakReferences` of Dart 2.17